packages/terra-pills/src/Pills.jsx

```diff
@@ -0,0 +1,385 @@
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const focusNodeAfterDelete = (pills, rollUpPill, pillCount, isRemovable) => {
+    if (isRemovable) {
+      // If there is only one pill and is removable
+      if (pills.length === 1 && rollUpPill === null) {
+        focusPillsContainer();
+        return;
+      }
+      if (pills.length - 1 < React.Children.count(children)) {
+        setTabIndex('-1');
+        if (focusNode.current === 0) { // If the first pill is deleted, focus the next pill
+          focusNode.current = 0;
+        } else {
+          focusNode.current -= 1;
+        }
+
+        isPillDeleted.current = true;
+        setRollUpCount(React.Children.count(children));
+        setUpdatedCount(React.Children.count(children));
+      }
+    }
+  };
+
+  const handlePillListKeyDown = (event) => {
+    const pills = [...pillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const pillCount = pills.length;
+    const rollUpPill = pillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const isPillRemovable = event.target.hasAttribute('data-terra-pill-removable');
+
+    switch (event.keyCode) {
+      case KEY_RIGHT:
+        event.preventDefault();
+        focusNextNode(pills, rollUpPill);
+        break;
+      case KEY_LEFT:
+        event.preventDefault();
+        focusPreviousNode(pills, rollUpPill);
+        break;
+      case KEY_BACK_SPACE || KEY_DELETE:
+        event.preventDefault();
+        focusNodeAfterDelete(pills, rollUpPill, pillCount, isPillRemovable);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handlePillListOnblur = () => setContainerTabindex(-1);
+
+  const handleOnRemove = (pillKey, metaData) => {
+    if (onRemove) {
+      onRemove(pillKey, metaData);
+    }
+  };
+
+  const handleOnSelectRollUp = (event) => {
+    if (onSelectRollUp) {
+      onSelectRollUp();
+    }
+
+    if (event.type === 'keydown') {
+      isRollUpRemoved.current = true;
+      setUpdatedCount(React.Children.count(children));
+    }
+  };
+
+  const handleWidthChange = () => {
+    if (isSingleLine) {
+      setUpdatedCount(React.Children.count(children));
+      setRollUpCount(React.Children.count(children));
+      generateRollUp();
+    }
+  };
+
+  const pillsProps = {};
+  pillsProps.onKeyDown = handlePillListKeyDown;
+  pillsProps.onBlur = handlePillListOnblur;
+
+  const pillListClassNames = classNames(
+    cx([
+      'pill-list',
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const renderChildren = (items) => {
+    const pills = React.Children.map(items, (pill) => {
+      if (React.isValidElement(pill)) {
+        return React.cloneElement(pill, { onRemove: handleOnRemove, isBasicPill: true });
+      }
+      return undefined;
+    });
+
+    const reducedArray = pills.splice(0, updatedCount);
+
+    return reducedArray;
+  };
+
+  return (
+    <ResponsiveElement responsiveTo="window" onResize={handleWidthChange}>
```

Comment:
You likely want to be responsive to "element" here. It's more expensive, but as-is, changes to the container size that aren't the result of the window size changing will not be handled.

Reply:
changed in this commit f2c63babe26a06cb3b0ee3ba1deb77728bd5a0ef